File: leetcode/solutions/0136.SingleNumber/singleNumber_test.ts
import { assertEquals } from "https://deno.land/std/testing/asserts.ts";
import singleNumber from "./singleNumber.ts";

Deno.test("0136. Single Number", () => {
  assertEquals(singleNumber([2, 2, 1]), 1);
  assertEquals(singleNumber([4, 1, 2, 1, 2]), 4);
});

File: leetcode/solutions/0136.SingleNumber/singleNumber_test.go
package solution

import (
	"testing"

	"github.com/google/go-cmp/cmp"
)

func Test_0136_SingleNumber(t *testing.T) {
	t.Parallel()
	tests := []struct {
		in  []int
		out int
	}{
		{
			[]int{2, 2, 1},
			1,
		},
		{
			[]int{4, 1, 2, 1, 2},
			4,
		},
	}

	for _, tt := range tests {
		tt := tt

		t.Run("Test", func(t *testing.T) {
			t.Parallel()

			got := singleNumber(tt.in)

			if !cmp.Equal(got, tt.out) {
				t.Errorf("got %#v want  %#v", got, tt.out)
			}
		})
	}
}
